tastypie/fields.py

```diff
@@ -763,6 +764,15 @@ def hydrate(self, bundle):
         if value is None:
             return value
 
+        if bundle.obj and isinstance(getattr(bundle.obj.__class__, self.attribute), SingleRelatedObjectDescriptor):
+            # This is the case when we are writing to a reverse one to one field.
+            # Enable related name to make this work fantastically.
+            # see https://code.djangoproject.com/ticket/18638 (bug; closed; worksforme)
+            # and https://github.com/toastdriven/django-tastypie/issues/566
+
+            # this gets the related_name of the one to one field of our model
+            self.related_name = getattr(bundle.obj.__class__, self.attribute).related.field.name
```

Comment:
Even if null=True I think the field instance should still be there. I should change it to not modify the field instance based on bundle data though, related_name should be set during __init__.

Opened #1415